cmd/test/main.go

```diff
@@ -303,7 +322,14 @@ func test(ctx *TestCommandContext, paths []string, options *TestCommandOptions)
 		ctx.Printer.SetTheme(printer.CreateSimpleTheme())
 	}
 
-	validationManager, createEvaluationResponse, results, err := evaluate(ctx, filesPaths, options)
+	policy, err := policy_factory.CreatePolicy(policies, options.PolicyName)
+	if err != nil {
+		return err
+	}
+
+	options.PolicyName = policy.Name
```

Comment:
`options` population should happen outside of `test` command. that way you will only forward to `test` the required policy